dynawo/sources/Solvers/Common/DYNSolverCommon.cpp

```diff
@@ -67,6 +72,51 @@ SolverCommon::copySparseToKINSOL(const SparseMatrix& smj, SUNMatrix JJ, const in
   return matrixStructChange;
 }
 
+void SolverCommon::propagateMatrixStructureChangeToKINSOL(const SparseMatrix& smj, SUNMatrix JJ, const int& size, sunindextype* lastRowVals,
+                                                          SUNLinearSolver LS, const std::string& linearSolverName, bool log) {
+  bool matrixStructChange = copySparseToKINSOL(smj, JJ, size, lastRowVals);
+
+  if (matrixStructChange) {
+    if (linearSolverName == "KLU") {
+      SUNLinSol_KLUReInit(LS, JJ, SM_NNZ_S(JJ), 2);  // reinit symbolic factorisation
+#ifdef WITH_NICSLU
+    } else if (linearSolverName == "NICSLU") {
+      SUNLinSol_NICSLUReInit(LS, JJ, SM_NNZ_S(JJ), 2);  // reinit symbolic factorisation
+    }
+#else
+  }
+#endif
+    if (lastRowVals != NULL) {
+      free(lastRowVals);
+    }
+    lastRowVals = reinterpret_cast<sunindextype*> (malloc(sizeof (sunindextype)*SM_NNZ_S(JJ)));
+    memcpy(lastRowVals, SM_INDEXVALS_S(JJ), sizeof (sunindextype)*SM_NNZ_S(JJ));
+    if (log)
+      Trace::debug() << DYNLog(MatrixStructureChange) << Trace::endline;
+  }
+}
+
+void
+SolverCommon::printLargestErrors(std::vector<std::pair<double, int> >& fErr, const boost::shared_ptr<Model>& model,
+                   int nbErr) {
+  std::sort(fErr.begin(), fErr.end(), mapcompabs());
```

Comment:
Would be more efficient to use a map, but as this is debug we can skip it if it induces too many changes

Reply:
What would be unique would be the int (index of the equation). The residual value (the double) could be non unique.
So the map key would have to be the index of the equation. As such, the map order wouldn't be the one I want to have here - I would like to keep it sorted by decreasing values to print the largest errors. 

So I get the impression that a map won't really help (btw, I don't know the motivation of the vector<pair>: I just copied and pasted it from the code I deleted in the first commit).

Reply:
map<int, double> then :)
It is just a detail as it is only used for in debug mode. We can drop it.